plugin/storage/es/factory.go

```diff
@@ -218,7 +218,7 @@ func (f *Factory) CreateArchiveSpanReader() (spanstore.Reader, error) {
 	if !f.archiveConfig.Enabled {
 		return nil, nil
 	}
-	sr, err := createSpanReader(f.getArchiveClient, f.archiveConfig, true, f.logger, f.tracer)
+	sr, err := createSpanReader(f.getArchiveClient, f.archiveConfig, f.logger, f.tracer)
```

Comment:
do we want to override the config here to set `MaxSpanAge` - or do we always respect what is set by the user

Reply:
I think you're starting from the wrong direction. I would first look into the actual code that uses `archive` flag and see if you can rewrite that code by relying on the other parameters that make sense. Then we can decide how to initialize those parameters.